components/ml/org.wso2.carbon.ml.rest.api/src/main/java/org/wso2/carbon/ml/rest/api/neuralNetworks/FeedForwardNetwork.java

```diff
@@ -0,0 +1,511 @@
+            int nInput = 0;
+            if(i == 0)
+                nInput=inputLayerNodes;
+            else
+                nInput=hiddenList.get(i-1).hiddenNodes;
+
+            neuralNetConfiguration.layer(i,new DenseLayer.Builder().nIn(nInput)
+                    .nOut(hiddenList.get(i).hiddenNodes)
+                    .weightInit(mapWeightInit(hiddenList.get(i).weightInit))
+                    .activation(hiddenList.get(i).activationAlgo)
+                    .build());
+        }
+
+        //Add Output Layers to the network with unique settings
+        neuralNetConfiguration.layer(noHiddenLayers, new OutputLayer.Builder(mapLossFunction(outputList.get(0).lossFunction))
+                    .nIn(hiddenList.get(noHiddenLayers-1).hiddenNodes)
+                    .nOut(outputList.get(0).outputNodes)
+                    .weightInit(mapWeightInit(outputList.get(0).weightInit))
+                    .activation(outputList.get(0).activationAlgo)
+                    .build());
+
+        //Create MultiLayerConfiguration network
+        MultiLayerConfiguration conf = neuralNetConfiguration.pretrain(pretrain)
+                .backprop(backprop).build();
+
+        MultiLayerNetwork model = new MultiLayerNetwork(conf);
+        model.init();
+        model.setListeners(Collections.singletonList((IterationListener) new ScoreIterationListener(1)));
+
+        while (trainIter.hasNext()) {
+            currentDataset = trainIter.next();
+            splitTestAndTrain = currentDataset.splitTestAndTrain(fraction,rnd);
+            trainingSet = splitTestAndTrain.getTrain();
+            testingSet = splitTestAndTrain.getTest();
+            features= testingSet.getFeatureMatrix();
+            labels = testingSet.getLabels();
+        }
+
+        //Train the model with the training data
+        for ( int n = 0; n < nepoches; n++) {
+            model.fit( trainingSet);
+        }
+
+        //Do the evaluations of the model including the Accuracy, F1 score etc.
+        log.info("Evaluate model....");
+        Evaluation eval = new Evaluation(outputList.get(0).outputNodes);
+        predicted = model.output(features,false);
+
+        eval.eval(labels, predicted);
+
+        evaluationDetails = "{\"Accuracy\":\""+eval.accuracy()+"\", \"Pecision\":\""+eval.precision()+"\",\"Recall\":\""+eval.recall()+"\",\"F1Score\":\""+eval.f1()+"\"}";
+        return evaluationDetails;
+
+    }
+
+    /**
+     * method to map user selected Optimazation Algorithm to OptimizationAlgorithm object.
+     * @param optimizationAlgorithms
+     * @return an OptimizationAlgorithm object.
+     */
+
+    OptimizationAlgorithm mapOptimizationAlgorithm(String optimizationAlgorithms){
+
+        OptimizationAlgorithm optimizationAlgo = null;
+
+        switch (optimizationAlgorithms){
+            case "Line_Gradient_Descent":
+                optimizationAlgo = OptimizationAlgorithm.LINE_GRADIENT_DESCENT;
+                break;
+
```

Comment:
Fixed in https://github.com/wso2/carbon-ml/pull/233/commits/275de09f616359250d4252a996d9ef255bc68def